File: server/src/hsogserver/meta/impl/CarMetaModel.cpp
#include "CarMetaModel.h"

using namespace taco;

CarMetaModel::CarMetaModel(
		const double &floorHeight, const std::string &steeringServoName, const std::string &mainMotorName)
	: _floorHeight(floorHeight), _steeringServoName(steeringServoName), _mainMotorName(mainMotorName)
{
}

CarMetaModel::~CarMetaModel()
{
}

const std::string &CarMetaModel::getSteeringServoName() const
{
	return _steeringServoName;
}

const std::string &CarMetaModel::getMainMotorName() const
{
	return _mainMotorName;
}

const double &CarMetaModel::getFloorHeight() const
{
	return _floorHeight;
}

IAxleConfig::ConstPtr CarMetaModel::getFrontAxle() const
{
	return _frontAxle;
}

IAxleConfig::ConstPtr CarMetaModel::getRearAxle() const
{
	return _rearAxle;
}

const std::vector<ISensorConfig::ConstPtr> &CarMetaModel::getGyroConfigs() const
{
	return _gyroConfigs;
}

const std::vector<ISensorConfig::ConstPtr> &CarMetaModel::getAccelerometerConfigs() const
{
	return _accelerometerConfigs;
}

const std::vector<ISensorConfig::ConstPtr> &CarMetaModel::getIMUConfigs() const
{
	return _imuConfigs;
}

const std::vector<IDistanceSensorConfig::ConstPtr> &CarMetaModel::getInfraRedConfigs() const
{
	return _infraRedConfigs;
}

const std::vector<IDistanceSensorConfig::ConstPtr> &CarMetaModel::getUltrasonicConfigs() const
{
	return _ultrasonicConfigs;
}

const std::vector<ISensorConfig::ConstPtr> &CarMetaModel::getRotationConfigs() const
{
	return _rotationConfigs;
}

const std::vector<ISensorConfig::ConstPtr> &CarMetaModel::getVoltageConfigs() const
{
	return _voltageConfigs;
}

const std::vector<ICameraConfig::ConstPtr> &CarMetaModel::getCameraConfigs() const
{
	return _cameraConfigs;
}

const std::vector<ICameraConfig::ConstPtr> &CarMetaModel::getDepthCameraConfigs() const
{
	return _depthCameraConfigs;
}

const std::vector<IServoDriveConfig::ConstPtr> &CarMetaModel::getServoDriveConfigs() const
{
	return _servoDriveConfigs;
}

const std::vector<IActuatorConfig::ConstPtr> &CarMetaModel::getMotorConfigs() const
{
	return _motorConfigs;
}

const std::vector<IActuatorConfig::ConstPtr> &CarMetaModel::getLightConfigs() const
{
	return _lightConfigs;
}

const std::vector<IActuatorConfig::ConstPtr> &CarMetaModel::getManeuverStatusConfigs() const
{
	return _statusConfigs;
}

File: client/src/taco/agent/model/agentmeta/impl/CarMetaModel.java
package taco.agent.model.agentmeta.impl;

import hso.autonomy.agent.model.agentmeta.impl.SensorConfiguration;

/**
 * The deserialized CarMetaModel.json file.
 */
public class CarMetaModel
{
	public static final String DEFAULT_DIRECTORY = "../config";

	private float floorHeight;

	private String steeringServoName;

	private String mainMotorName;

	private AxleConfiguration frontAxle;

	private RearAxleConfiguration rearAxle;

	private MountedSensorConfiguration[] imuConfigs;

	private DistanceSensorConfiguration[] ultrasonicConfigs;

	private MountedSensorConfiguration[] rotationConfigs;

	private SensorConfiguration[] voltageConfigs;

	private CameraConfiguration[] cameraConfigs;

	private CameraConfiguration[] depthCameraConfigs;

	private SensorConfiguration[] motorConfigs;

	private SensorConfiguration[] lightConfigs;

	private SensorConfiguration[] statusConfigs;

	private ServoDriveConfiguration[] servoDriveConfigs;

	public float getFloorHeight()
	{
		return floorHeight;
	}

	public String getSteeringServoName()
	{
		return steeringServoName;
	}

	public String getMainMotorName()
	{
		return mainMotorName;
	}

	public AxleConfiguration getFrontAxle()
	{
		return frontAxle;
	}

	public RearAxleConfiguration getRearAxle()
	{
		return rearAxle;
	}

	public MountedSensorConfiguration[] getImuConfigs()
	{
		return imuConfigs;
	}

	public DistanceSensorConfiguration[] getUltrasonicConfigs()
	{
		return ultrasonicConfigs;
	}

	public MountedSensorConfiguration[] getRotationConfigs()
	{
		return rotationConfigs;
	}

	public SensorConfiguration[] getVoltageConfigs()
	{
		return voltageConfigs;
	}

	public CameraConfiguration[] getCameraConfigs()
	{
		return cameraConfigs;
	}

	public CameraConfiguration[] getDepthCameraConfigs()
	{
		return depthCameraConfigs;
	}

	public SensorConfiguration[] getMotorConfigs()
	{
		return motorConfigs;
	}

	public SensorConfiguration[] getLightConfigs()
	{
		return lightConfigs;
	}

	public SensorConfiguration[] getStatusConfigs()
	{
		return statusConfigs;
	}

	public ServoDriveConfiguration[] getServoDriveConfigs()
	{
		return servoDriveConfigs;
	}

	public double getAxleSpacing()
	{
		return getFrontAxle().getPosition().getX() - getRearAxle().getPosition().getX();
	}
}
